### PR TITLE
HOTT-3166 Garbage collect wizard sessions

### PR DESCRIPTION
**Pull request by jebw**
### Jira link

HOTT-3166

### What?

I have added/removed/altered:

- [x] Added a custom wizard store which garbage collects old journeys

### Why?

I am doing this because:

- We are overflowing the cookie size because our sessions are long lived

### Deployment risks (optional)

- Removes user session data

### CODE REVIEW

File: app/models/rules_of_origin/wizard_store.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        backing_store.keys
+                     .select { |k| k.start_with? PREFIX }
+                     .sort_by { |k| backing_store[k]['modified'] || '2022-01-01T00:00:00Z' }
+                     .reverse
+                     .slice(MAX_JOURNEYS..)
+                     &.each { |k| backing_store.delete k }
+      end
+    end
+
+    def initialize(backing_store, key)
```
Comment (willfish) [inline app/models/rules_of_origin/wizard_store.rb:17]: Shouldn't this be in the normal scope of the class?

File: app/models/rules_of_origin/wizard_store.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    MAX_JOURNEYS = 5
+
+    class << self
+      def gc_store_data(backing_store)
+        backing_store.keys
+                     .select { |k| k.start_with? PREFIX }
+                     .sort_by { |k| backing_store[k]['modified'] || '2022-01-01T00:00:00Z' }
+                     .reverse
+                     .slice(MAX_JOURNEYS..)
+                     &.each { |k| backing_store.delete k }
```
Comment (willfish) [inline app/models/rules_of_origin/wizard_store.rb:13]: I suspect & isn't needed, here?

**Review by willfish — APPROVED** [review]
LGTM

File: app/models/rules_of_origin/wizard_store.rb (hunk 2 above)
Comment (jebw) [reply app/models/rules_of_origin/wizard_store.rb:13]: :eyes:

File: app/models/rules_of_origin/wizard_store.rb (hunk 1 above)
Comment (jebw) [reply app/models/rules_of_origin/wizard_store.rb:17]: after discussion its not an issue

File: app/models/rules_of_origin/wizard_store.rb (hunk 2 above)
Comment (jebw) [reply app/models/rules_of_origin/wizard_store.rb:13]: Unfortunately slice returns nil, not an empty array if no matches

```
Loading development environment (Rails 7.0.4.3)
[1] pry(main)> [1,2,3,4].slice(5..)
=> nil
[2] pry(main)> 
```

**Pull request merged**